### DISCUSSION TITLE
Obtaining a sequence using OR

### DISCUSSION THREAD

**Discussion by eposha** [prisma/prisma#20666] [Q&A] [answered]
### Question

Is it possible to retrieve data in a strict sequence passed to OR with a single query?

For example I need to retrieve 50 records in total, the sequence should be like this, first by city, then by state, then by country

I have used OR, but the values are returned in different sequence

### Information about Prisma Schema, Client Queries and Environment (optional)

```ts
 OR: [
           { city: { path: ['id'], equals: city } },
           {
             city: { path: ['id'], not: city },
             state: { path: ['id'], equals: stateId },
            },
            {
              city: { path: ['id'], not: city },
              state: { path: ['id'], not: stateId },
              country: { not: '' },
             },
        ]
```

**Comment by ludralph** [comment]
Hi @eposha 👋 

Using the OR condition in your query might not guarantee a strict sequence of results as you've described. The database management system (DBMS) might optimize the query execution and return the results in a different order. If you need a specific sequence of results based on multiple conditions, it's better to sort the results after retrieving them using the desired criteria. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by eposha** [comment, marked as answer]
@ludralph Hi, thanks for the answer

I found another solution
